framework/aster-frame/src/sync/wait.rs

```diff
@@ -35,72 +29,50 @@ impl WaitQueue {
     /// The method only returns if the condition returns Some(_).
     /// A waker thread should first make the condition Some(_), then invoke the
     /// `wake`-family method. This ordering is important to ensure that waiter
-    /// threads do not lose any wakeup notifiations.
+    /// threads do not lose any wakeup notifications.
     ///
     /// By taking a condition closure, his wait-wakeup mechanism becomes
     /// more efficient and robust.
-    pub fn wait_until<F, R>(&self, cond: F) -> R
+    pub fn wait_until<F, R>(&self, mut cond: F) -> R
     where
         F: FnMut() -> Option<R>,
     {
-        self.do_wait(cond, None).unwrap()
-    }
+        if let Some(res) = cond() {
+            return res;
+        }
 
-    /// Wait until some condition returns Some(_), or a given timeout is reached. If
-    /// the condition does not becomes Some(_) before the timeout is reached, the
-    /// function will return None.
-    pub fn wait_until_or_timeout<F, R>(&self, cond: F, timeout: &Duration) -> Option<R>
-    where
-        F: FnMut() -> Option<R>,
-    {
-        self.do_wait(cond, Some(timeout))
+        let (waiter, _) = Waiter::new_pair();
+
+        self.wait_until_or_cancelled(cond, waiter, || false)
+            .unwrap()
     }
 
-    fn do_wait<F, R>(&self, mut cond: F, timeout: Option<&Duration>) -> Option<R>
+    /// Wait until some condition becomes true or the cancel condition becomes true.
+    ///
+    /// This method will returns Some(_) if the condition returns Some(_), and will returns
```

Comment:
- `will returns` -> `will return`
 - `Some(_)` -> `` `Some(_)` ``

Perhaps also fix comments of other methods in `WaitQueue`.